app/src/main/java/org/schabi/newpipe/database/history/dao/SearchHistoryDAO.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-FileCopyrightText: 2017-2021 NewPipe contributors <https://newpipe.net>
+ * SPDX-FileCopyrightText: 2025 NewPipe e.V. <https://newpipe-ev.de>
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+package org.schabi.newpipe.database.history.dao
+
+import androidx.room.Dao
+import androidx.room.Query
+import io.reactivex.rxjava3.core.Flowable
+import org.schabi.newpipe.database.history.model.SearchHistoryEntry
+
+@Dao
+interface SearchHistoryDAO : HistoryDAO<SearchHistoryEntry> {
+
+    @get:Query("SELECT * FROM search_history WHERE id = (SELECT MAX(id) FROM search_history)")
```

Comment:
What is the rationale for writing queries as full strings instead of as concatenations of reusable pieces? I am fine with this change as it makes things simpler, but it also might increase the chance of misspelling something (or does room catch that?). So yeah please give some rationale :-)

Reply:
Writing queries as full or multi-line strings allows us to use IDE hints and lint which isn't available with concatenations. My base for choosing this was based on the existing kotlin DAO classes which also use full strings instead of concatenations.